src/python/bot/tasks/impact_task.py

```diff
@@ -242,7 +242,16 @@ def get_impacts_on_prod_builds(testcase, testcase_file_path):
 def get_impact_on_build(build_type, current_version, testcase,
                         testcase_file_path):
   """Return impact and additional trace on a prod build given build_type."""
-  build = build_manager.setup_production_build(build_type)
+  # TODO(yuanjunh): remove es_enabled var after testing is done.
+  es_enabled = testcase.get_metadata('es_enabled', False)
+  if build_type == 'extended_stable' and not es_enabled:
+    return Impact()
+  try:
+    build = build_manager.setup_production_build(build_type)
+  except Exception as e:
```

Comment:
still keep try-except in case any exceptions in setting up extended stable build.